test/Microsoft.ML.Tests/TrainerEstimators/TrainerEstimators.cs

```diff
@@ -82,20 +83,62 @@ public void KMeansEstimator()
         }
 
         /// <summary>
-        /// HogwildSGD TrainerEstimator test 
+        /// HogwildSGD TrainerEstimator test (logistic regression).
         /// </summary>
         [Fact]
         public void TestEstimatorHogwildSGD()
         {
             (IEstimator<ITransformer> pipe, IDataView dataView) = GetBinaryClassificationPipeline();
-            var trainer = ML.BinaryClassification.Trainers.StochasticGradientDescent();
-            var pipeWithTrainer = pipe.Append(trainer);
-            TestEstimatorCore(pipeWithTrainer, dataView);
+            var trainers = new[] { ML.BinaryClassification.Trainers.StochasticGradientDescent(l2Weight:0),
+                ML.BinaryClassification.Trainers.StochasticGradientDescent(new Trainers.SgdBinaryTrainer.Options(){ L2Weight=0 })};
+
+            foreach (var trainer in trainers)
+            {
+                var pipeWithTrainer = pipe.Append(trainer);
+                TestEstimatorCore(pipeWithTrainer, dataView);
+
+                var transformedDataView = pipe.Fit(dataView).Transform(dataView);
+                var model = trainer.Fit(transformedDataView);
+                trainer.Fit(transformedDataView, model.Model);
+                TestEstimatorCore(pipe, dataView);
+
+                var result = model.Transform(transformedDataView);
+                var metrics = ML.BinaryClassification.EvaluateNonCalibrated(result);
```

Comment:
>EvaluateNonCalibrated [](start = 54, length = 21)

Doesn't this one produce calibrated model?

Reply:
Right. Will test logloss too.

---
In reply to: [257351073](https://github.com/dotnet/machinelearning/pull/2564#discussion_r257351073) [](ancestors = 257351073)